articles/native-platforms/android/03-session-handling.md

```diff
@@ -0,0 +1,111 @@
+---
+title: Session Handling
+description: This tutorial will show you how to use Lock10 to mantain a session connected.
+---
+
+## Android - Session Handling Tutorial
+
+This tutorial will show you how to use Lock10 to mantain an active session with Auth0.
+
+The user's ``credentials`` class contain three kind of tokens:
+
+* ``idToken``: Identity Token that proves the identity of the user.
+* ``accessToken``: Access Token used by Auth0 API.
+* ``refreshToken``: Refresh Token that can be used to request new tokens without signing in again.
+
+::: panel-info System Requirements
+This tutorial and seed project have been tested with the following:
+
+* AndroidStudio 2.0
+* Emulator - Nexus5X - Android 6.0 
+  :::
+
+
+### Before Starting
+
+Be sure that you have completed the [Basic Login](01-login.md) Quickstart.
+
+### 1. Save the User's Credentials
+
+Your first step is to save, through a secure method, the user's credentials obtained in the login success response. We won't cover how to do that on this tutorial, feel free to save it as you like.
+
+```java
+private LockCallback callback = new AuthenticationCallback() {
+            @Override
+            public void onAuthentication(Credentials credentials) {
+				// Login Success response
+				saveCredentials(credentials)
+            }
+				...
+        };
+```
+
+ 
+### 2. Request a new tokenID
+
+#### i. Using a non-expired tokenID
+
+In the case your current tokenID didn't expire, you can use it to get a new one, with a newer expire date.
+```java
```

Comment:
@tenmaster Please remove excessive tabs at the beginning of lines, from all scripts in this doc. For example:

![image](https://cloud.githubusercontent.com/assets/11715799/16986761/fb2f89fc-4e90-11e6-973d-56ffe13d91b2.png)